utils/automatic-release/process.js

```diff
@@ -0,0 +1,121 @@
+'use strict';
+
+const spawn = require('cross-spawn');
+const fs = require('fs-extra');
+const path = require('path');
+const yaml = require('yamljs');
+const differ = require('jest-diff');
+const diffToSemver = require('./diffToSemver');
+const console = require('console');
+
+// run lernaUpdate and return a list of the packages that need to be updated
+const root = path.join(__dirname, '../../');
+const latestTemp = path.join(__dirname, './tempNpm');
+
+const checkUPdated = () => {
+  const cmd = spawn('npm', ['run', 'updated'], {
+    cwd: root
+  });
+  let result = '';
+  cmd.stdout.on('data', function(data) {
+    result += data.toString();
+  });
+
+  return new Promise(resolve => {
+    cmd.on('close', code => {
+      return resolve(
+        result
+          .split('\n')
+          .filter(line => line.match(/^-\sot/))
+          .map(line => line.replace(/^-\s/, ''))
+      );
+    });
+  });
+};
+
+// locally install latest version of packages
+const installLatestFromNPM = packages => {
+  fs.ensureDirSync(latestTemp);
+  fs.writeJsonSync(`${latestTemp}/package.json`, { name: 'release' });
+
+  const installed = packages.map(pkg => {
+    const cmd = spawn('npm', ['install', pkg], {
+      stdio: 'inherit',
+      cwd: latestTemp
+    });
+    return new Promise(resolve => {
+      cmd.on('close', code => (code === 0 ? resolve(pkg) : resolve(null)));
+    });
+  });
+
+  return installed;
+};
+
+const run = () => {
+  return new Promise((resolve, reject) => {
+    checkUPdated()
+      .then(updated => {
+        Promise.all(installLatestFromNPM(updated)).then(() => {
+          updated.forEach(pkg => {
+            const updatedPath = pkg.match(/^ottheme/)
+              ? `OTTheme/${pkg}/token.yml`
+              : `OTKit/${pkg}/token.yml`;
```

Comment:
This name selector seems a bit specific to our use case. Can we move this path resolver at least to a separate file?
Perhaps some kind of `conf.js` like
```js
module.exports.pathResolver = pkg => ...
```
So that in the future this is a bit easier to spot/change/configure?

Reply:
Very good idea, I likes it, thanks for it!